website/content/v1.7/talos-guides/install/cloud-platforms/aws.md

```diff
@@ -106,228 +156,329 @@ Once the `SnapshotTaskDetail.Status` indicates `completed`, we can register the
+            "DeviceIndex":0,
+            "AssociatePublicIpAddress":true,
+            "Groups":["'$SECURITY_GROUP_ID'"],
+            "DeleteOnTermination":true
+        }],
+        "BlockDeviceMappings":[{
+            "DeviceName":"/dev/xvda",
+            "Ebs":{
+                "VolumeSize":20,
+                "VolumeType":"gp3",
+                "DeleteOnTermination":true
+            }
+        }],
+        "TagSpecifications":[{
+            "ResourceType":"instance",
+            "Tags":[{
+          "Key":"Name",
+          "Value":"talos-aws-tutorial-worker"
+          }]
+        }]}' \
+    --query 'LaunchTemplate.LaunchTemplateId' \
+    --output text)
+```
+
+```bash
+aws autoscaling create-auto-scaling-group \
+    --auto-scaling-group-name talos-aws-tutorial-worker \
+    --min-size 1 \
+    --max-size 3 \
+    --desired-capacity 1 \
+    --availability-zones $(echo ${AZS[@]}) \
+    --target-group-arns $TARGET_GROUP_ARN \
+    --launch-template "LaunchTemplateId=${WORKER_LAUNCH_TEMPLATE_ID}" \
+    --vpc-zone-identifier $(echo ${SUBNETS[@]} | tr ' ' ',')
 ```
 
 ### Configure the Load Balancer
 
 Now, using the load balancer target group's ARN, and the **PrivateIpAddress** from the controlplane instances that you created :
 
 ```bash
-aws elbv2 register-targets \
-    --region $REGION \
+for INSTANCE in ${CP_INSTANCES[@]}; do
+    aws elbv2 register-targets \
     --target-group-arn $TARGET_GROUP_ARN \
-    --targets Id=$CP_NODE_1_IP  Id=$CP_NODE_2_IP  Id=$CP_NODE_3_IP
+    --targets Id=$(aws ec2 describe-instances \
+        --instance-ids $INSTANCE \
+        --query 'Reservations[].Instances[].InstanceId' \
+        --output text)
+done
 ```
 
-Using the ARNs of the load balancer and target group from previous steps, create the listener:
+### Bootstrap `etcd`
+
+Export the `talosconfig` file so commands sent to Talos will be authenticated.
 
 ```bash
-aws elbv2 create-listener \
-    --region $REGION \
-    --load-balancer-arn $LOAD_BALANCER_ARN \
-    --protocol TCP \
-    --port 443 \
-    --default-actions Type=forward,TargetGroupArn=$TARGET_GROUP_ARN
-```
+export TALOSCONFIG=$(pwd)/talosconfig
 
-### Bootstrap Etcd
+WORKER_INSTANCES=( $(aws autoscaling \
+    describe-auto-scaling-instances \
+    --query 'AutoScalingInstances[?AutoScalingGroupName==`talos-aws-tutorial-worker`].InstanceId' \
+    --output text) )
+```
 
 Set the `endpoints` (the control plane node to which `talosctl` commands are sent) and `nodes` (the nodes that the command operates on):
 
 ```bash
-talosctl --talosconfig talosconfig config endpoint <control plane 1 PUBLIC IP>
-talosctl --talosconfig talosconfig config node <control plane 1 PUBLIC IP>
+talosctl config endpoints $(aws ec2 describe-instances \
+    --instance-ids $(echo $CP_INSTANCES[@]) \
+    --query 'Reservations[].Instances[].PublicIpAddress' \
+    --output text)
+
+talosctl config nodes $(aws ec2 describe-instances \
+    --instance-ids $(echo $CP_INSTANCES[1]) \
```

Comment:
I didn't actually execute anything past here, since I knew how to grab this content, but again with the `{}` for your arrays:
```suggestion
    --instance-ids $(echo ${CP_INSTANCES[1]}) \
```